src/spider/scheduler/FifoPolicy.cpp

```
@@ -57,89 +60,76 @@ auto task_locality_satisfied(
-            }
-    );
-
-    return earliest_task->get_id();
+    m_tasks.erase(it.base());
+    for (core::TaskInput const& input : it->get_inputs()) {
+        std::optional<boost::uuids::uuid> const data_id = input.get_data_id();
+        if (data_id.has_value()) {
+            m_data_cache.erase(data_id.value());
+        }
+    }
+    return it->get_id();
```

Cppcheck (2.10-2)

[performance] Variable 'id' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)